src/debug-adapter/client.ts

```diff
@@ -209,10 +209,20 @@ class BazelDebugSession extends DebugSession {
     response: DebugProtocol.SetBreakpointsResponse,
     args: DebugProtocol.SetBreakpointsArguments,
   ) {
-    // VS Code passes us the absolute path to the file, but Bazel's debugger
-    // will match against paths beneath the output_base. It appears that even
-    // .bzl files in the same workspace as the one containing the target being
-    // built will be found in the "external" subtree.
+    // The path we need to pass to Bazel here depends on how the .bzl file has
+    // been loaded. Unfortunately this means we have to create two breakpoints,
+    // one for each possible path, because the way the .bzl file is loaded is
+    // chosen by the user:
+    //
+    // 1. If the file is loaded using an explicit repository reference (i.e.,
+    //    `@foo//:bar.bzl`), then it will appear in the `external` subdirectory
+    //    of Bazel's output_base.
+    // 2. If the file is loaded as a same-repository path (i.e., `//:bar.bzl`),
```

Comment:
I thought you had said something also about these symlinks not existing until the build is started.  Does that play into anything you have to deal with?

Reply:
That's only an issue if I need to _resolve_ them, which I'm not doing here; I'm just constructing the paths (or in the case of the new path, using the one I already have).